### PR TITLE
elementBodyResponse in combination with doNotGroupByColPos = 1

### PR DESCRIPTION
**Pull request by MohsinQK**
Fixed the possibility to get elementBodyResponse also if doNotGroupByColPos = 1 is set on lib.content

### CODE REVIEW

File: Classes/Middleware/ElementBodyResponseMiddleware.php [hunk 1]
```diff
@@ -106,6 +106,30 @@ private function extractElement(array $content, int $elementId, bool $recursiveE
...
         }
+        
+        // If no content found then its possible that option doNotGroupByColPos = 1 is set
+        // then try to find without groups of colpos
+        foreach ($content as $item) {
+            if (!is_array($item)) {
+                continue;
+            }
+
+            if ((int)($item['id'] ?? 0) === $elementId) {
```
Comment (twoldanski) [inline Classes/Middleware/ElementBodyResponseMiddleware.php:117]: I think we can add this condition in first iteration in content to check if we have flat array

Comment (twoldanski) [reply Classes/Middleware/ElementBodyResponseMiddleware.php:117]: and then we can skip whole additional iterations

**Review by twoldanski — CHANGES_REQUESTED** [review]

File: Classes/Middleware/ElementBodyResponseMiddleware.php [hunk 2]
```diff
@@ -84,6 +84,11 @@ private function extractElement(array $content, int $elementId, bool $recursiveE
         $body = [];
 
         foreach ($content as $items) {
+            // skip iteration if array is flat means doNotGroupByColPos = 1 is set
+            if(isset($items['id'])) {
```
Comment (twoldanski) [inline Classes/Middleware/ElementBodyResponseMiddleware.php:88]: here you can just in use (int)($items['id'] ?? 0) === $elementId if true, array is flat & match id, then return $items, else search in nested colPos, just then condition should be after is_array check, then you can drop whole extra foreach after main one

**Review by twoldanski — APPROVED** [review]

File: Classes/Middleware/ElementBodyResponseMiddleware.php [hunk 3]
```diff
@@ -84,6 +84,11 @@ private function extractElement(array $content, int $elementId, bool $recursiveE
         $body = [];
 
         foreach ($content as $items) {
+            // if array is flat means doNotGroupByColPos = 1 is set
```
Comment (twoldanski) [inline Classes/Middleware/ElementBodyResponseMiddleware.php:87]: this check should be after is_array

**Review by twoldanski — CHANGES_REQUESTED** [review]

**Review by lukaszuznanski — APPROVED** [review]

**Review by twoldanski — APPROVED** [review]

**Pull request merged**